packages/netdog/write-network-status.service

```
@@ -1,14 +1,16 @@
 [Unit]
 Description=Write network status
+DefaultDependencies=no
 # Block manual interactions with this service, since it could leave the system in an
 # unexpected state
 RefuseManualStart=true
 RefuseManualStop=true
 Before=early-boot-config.service
 # This service creates a symlink to the resolv.conf systemd-resolved creates
 # and we would like it to exist first
-After=systemd-networkd-wait-online.service systemd-resolved.service
-Wants=systemd-networkd-wait-online.service systemd-resolved.service
+RequiresMountsFor=/.bottlerocket
+After=systemd-networkd-wait-online.service systemd-resolved.service run-netdog.mount prepare-sysctl.service
+Requires=systemd-networkd-wait-online.service systemd-resolved.service run-netdog.mount prepare-sysctl.service
```

To avoid the change to `run-netdog.mount`:
```suggestion
RequiresMountsFor=/.bottlerocket /run/netdog
After=systemd-networkd-wait-online.service systemd-resolved.service prepare-sysctl.service
Requires=systemd-networkd-wait-online.service systemd-resolved.service prepare-sysctl.service
```

I would probably also just add the `mkdir -p /etc/sysctl.d` as another ExecStart command, vs. adding another unit that systemd has to do the bookkeeping for.
The `run-netdog.mount` change is a little odd because that unit is already `WantedBy` `network-pre.target`, which is reached before any of these

---

I think systemd ordering phase, `/run/netdog` gets mounted pretty early. This is just to **explicitly** call out the dependency here since we now use /run/netdog for the status files. IMO it does not hurt. Thoughts?